src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -369,8 +368,24 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
         database = it->second;
     }
 
-    auto table = database->tryGetTable(table_id.table_name, context_);
-    if (!table && exception)
+    StoragePtr table;
+    if (exception)
+    {
+        try
+        {
+            table = database->getTable(table_id.table_name, context_);
+        }
+        catch (const Exception & e)
+        {
+            exception->emplace(e);
+        }
+    }
+    else
+    {
+        table = database->tryGetTable(table_id.table_name, context_);
+    }
+
+    if (!table && exception && !exception->has_value())
```

Comment:
what difference it makes? do we expect `table` can be empty but no exception thrown at 376 so we need 388?
we need it to modify in #49801

Reply:
There could be different exceptions in obtaining a table, such as: - it does not exist; - the file format is unknown.

The latter is new after the introduction of DatabaseFilesystem.

This means that if we resort to always calling `tryGetTable`, we can lose precise information about the exception.